src/components/Nav/NavItem/NavItem.js

```diff
@@ -120,20 +131,21 @@ export default class NavItem extends Component {
     return (
       <li
         className={classNames}
+        label={label}
         onClick={event => handleDisabled(onClick(event, href))}
         onKeyPress={event => handleDisabled(onClick(event, href))}
         role="menuitem"
         tabIndex={handleDisabled(children ? -1 : tabIndex)}
-        {...other}
       >
         {link ? (
-          <a
+          <NavItemLink
             id={id}
             className={classnames(linkClassName, {
```

Comment:
- [ ] It would make sense for these classes to be part of the `NavItemLink` component

Reply:
Unfortunately that would also involve moving all of the external link + icon logic into the `NavItemLink` component as well 🤔 Could we keep these classes here, since `NavItemLink` is utilized directly by `NavItem` (and not necessarily defined by the user?)

Reply:
Fair point - if it's specific to the `NavItem`, my gut would be to bring all of that logic with it.

However, it's more likely that this logic could be used in links elsewhere throughout the `Shell`, I would recommend moving and renaming it to something like `ShellLink`, and that way we could start to use it in other places like the `Header` in future. What are your thoughts?

Reply:
That's an interesting approach but but I'm not sure where else it would be used in the shell... 😅 where else are "items" defined (except for the inside the `Nav` in the `Toolbar`)? Everything else (like notifications) uses an object & the "list" isn't necessarily exposed. The `Nav` is the only place this type of thing happens.

Reply:
I don't see an inherit "risk" in creating this component because it's just something used by `NavItem`. It wouldn't necessarily make a `Shell` refactor difficult or impossible. It's just taking some logic out of `NavItem.js` for the forwardRef + custom element.